x-pack/plugins/maps/public/classes/styles/vector/symbol_utils.js

```diff
@@ -41,13 +43,71 @@ export const SYMBOL_OPTIONS = Object.keys(SYMBOLS).map((symbolId) => {
   };
 });
 
+/**
+ * Converts a SVG icon to a monochrome image using a signed distance function.
+ *
+ * @param {string} svgString - SVG icon as string
+ * @param {number} [cutoff=0.25] - balance between SDF inside 1 and outside 0 of glyph
+ * @param {number} [radius=0.25] - size of SDF around the cutoff as percent of output icon size
+ * @return {ImageData} Monochrome image that can be added to a MapLibre map
+ */
+export async function createSdfIcon(svgString, cutoff = 0.25, radius = 0.25) {
+  const buffer = 3;
+  const size = 16 + buffer * 4;
+  const svgCanvas = document.createElement('canvas');
+  svgCanvas.width = size;
+  svgCanvas.height = size;
+  const svgCtx = svgCanvas.getContext('2d');
+  const v = Canvg.fromString(svgCtx, svgString, {
+    ignoreDimensions: true,
+    offsetX: buffer / 2,
+    offsetY: buffer / 2,
+  });
+  v.resize(size - buffer, size - buffer);
+  await v.render();
+
+  const distances = calcSDF(svgCtx, {
+    channel: 3,
+    cutoff,
+    radius: radius * size,
+  });
+
+  const canvas = document.createElement('canvas');
+  canvas.width = size;
+  canvas.height = size;
+  const ctx = canvas.getContext('2d');
+
+  const imageData = ctx.createImageData(size, size);
+  for (let i = 0; i < size; i++) {
+    for (let j = 0; j < size; j++) {
+      imageData.data[j * size * 4 + i * 4 + 0] = 0;
+      imageData.data[j * size * 4 + i * 4 + 1] = 0;
+      imageData.data[j * size * 4 + i * 4 + 2] = 0;
+      imageData.data[j * size * 4 + i * 4 + 3] = distances[j * size + i] * 255;
+    }
+  }
+  return imageData;
+}
+
 export function getMakiSymbolSvg(symbolId) {
   if (!SYMBOLS[symbolId]) {
     throw new Error(`Unable to find symbol: ${symbolId}`);
   }
   return SYMBOLS[symbolId];
 }
 
+export async function loadMakiIconInMap(symbolId, mbMap) {
+  const pixelRatio = Math.floor(window.devicePixelRatio);
```

Comment:
How about moving pixelRatio to within the scope of if statement. syncIconWithMb, where loadMakiIconInMap is called, gets [called any time layerList state changes](https://github.com/elastic/kibana/blob/main/x-pack/plugins/maps/public/connected_components/mb_map/mb_map.tsx#L344) (which is a lot). So avoiding any memory creation when the image already exists will help reduce GC.